lib/Controller/SAMLController.php

```diff
@@ -244,8 +264,22 @@ public function getMetadata($idp) {
 	 * @throws ValidationError
 	 */
 	public function assertionConsumerService() {
-		$AuthNRequestID = $this->session->get('user_saml.AuthNRequestID');
-		$idp = $this->session->get('user_saml.Idp');
+		// Fetch and decrypt the cookie
+		$cookie = $this->request->getCookie('saml_data');
+		if ($cookie === null) {
+			return;
+		}
+
+		// Base64 decode
+		$cookie = base64_decode($cookie);
+
+		// Decrypt and deserialize
+		$cookie = $this->crypto->decrypt($cookie);
```

Comment:
Mind to update the throws block? `decrypt` might throw a `\Exception`. I guess that could happen if one sends a self crafted saml_data cookie ;)

Reply:
Good catch yes. We should just catch it I guess and then return an error

Reply:
🏓